### PR TITLE
Fix: Use Snowflake's API to write a dataframe into a table

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/core/engine_adapter/base.py [hunk 1]
```diff
@@ -263,17 +263,14 @@ def _create_table_from_df(
         replace: bool = False,
         **kwargs: t.Any,
     ) -> None:
-        table = exp.to_table(table_name)
         if not isinstance(df, pd.DataFrame):
             raise ValueError("df must be a pandas DataFrame")
-        expression = next(
```
Comment (izeigerman) [inline sqlmesh/core/engine_adapter/base.py:269]: This logic was wrong. If he number of rows exceeded the `DEFAULT_BATCH_SIZE` of the adapter we'd write partial data.

File: sqlmesh/core/engine_adapter/snowflake.py [hunk 2]
```diff
@@ -64,3 +65,40 @@ def _get_data_objects(
...
+
+    def _insert_append_pandas_df(
+        self,
+        table_name: TableName,
+        df: pd.DataFrame,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+    ) -> None:
+        from snowflake.connector.pandas_tools import write_pandas
+
+        table = exp.to_table(table_name)
```
Comment (tobymao) [inline sqlmesh/core/engine_adapter/snowflake.py:77]: this, *db = tuple(reverse(part.name if part.quoted part.name.upper() for part in table.parts))

maybe?

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/engine_adapter/snowflake.py (hunk 2 above)
Comment (izeigerman) [reply sqlmesh/core/engine_adapter/snowflake.py:77]: nice!

**Review by eakmanrq — APPROVED** [review]
Nice catch on the bug in base. Love the more native implementations of handling Dataframes.

**Pull request merged**